cumulus_library/builders/protected_table_builder.py

```diff
@@ -64,12 +67,22 @@ def prepare_queries(
                 TRANSACTION_COLS_TYPES,
             )
         )
-        if manifest._study_config.get("statistics_config"):
-            self.queries.append(
-                base_templates.get_ctas_empty_query(
-                    db_schema,
-                    statistics,
-                    STATISTICS_COLS,
-                    STATISTICS_COLS_TYPES,
-                )
-            )
+        files = manifest.get_file_list()
+        files = [file for file in files if file.endswith(".toml")]
```

Comment:
This method works a bit differently than I'd expect - it looks through the file list itself, finds the first statistics type and then returns after creating tables for the first one it finds.

Why isn't this done here and not as part of a workflow (and thus part of the normal walk-through-the-file-list sequence), inside PsmBuilder or whatever?

Reply:
So what this is doing is creating a table that statistic type runs can write to, if it doesn't exist - the idea being that if there is more than one set of statistics, they'll all have one place to go - and also that this table will persist between individual study runs. It's basically a registry of all the sampled statistics.

I could add a clarifying comment?